src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingPlantation.java

```diff
@@ -88,315 +77,308 @@ public BuildingPlantation(final IColony c, final BlockPos l)
-            {
-                return false;
-            }
-            final Optional<Boolean> isRecipeAllowed = CraftingUtils.isRecipeCompatibleBasedOnTags(recipe, CRAFTING_PLANTATION);
-            return isRecipeAllowed.orElse(false);
+            return colony.getBuildingManager().getFreeFields(FieldRegistries.plantationField.get());
         }
 
         @Override
-        public @NotNull List<IGenericRecipe> getAdditionalRecipesForDisplayPurposesOnly()
+        public boolean canAddField(IField field)
         {
-            final List<IGenericRecipe> recipes = new ArrayList<>(super.getAdditionalRecipesForDisplayPurposesOnly());
-
-            // indicate which plants we can grow
-            for (final Map.Entry<Item, PlantationItem> entry : COMBINATIONS.entrySet())
+            if (field instanceof PlantationField plantationField)
             {
-                recipes.add(new GenericRecipe(null,
-                        new ItemStack(entry.getKey(), entry.getValue().getMinimumLength() + 1),
-                        Collections.emptyList(),
-                        Collections.singletonList(Collections.singletonList(new ItemStack(entry.getKey()))),
-                        1, Blocks.AIR, null, ToolType.NONE, Collections.emptyList(), -1));
+                final PlantationModule module = PlantationModuleRegistry.getPlantationModule(plantationField.getPlantationFieldType());
+                if (module != null && module.getRequiredResearchEffect() != null)
+                {
+                    return building.getColony().getResearchManager().getResearchEffects().getEffectStrength(module.getRequiredResearchEffect()) > 0;
+                }
+                return true;
             }
-
-            return recipes;
+            return false;
         }
     }
 
-    public static class PlantationSoilPosition
+    /**
+     * Field module view implementation for the plantation.
+     */
+    public static class PlantationFieldsModuleView extends FieldsModuleView
     {
         /**
-         * The position.
+         * The maximum amount of concurrent plants the planter can work on.
          */
-        private final BlockPos position;
+        private int maxConcurrentPlants = 0;
 
-        /**
-         * The item combination data.
-         */
-        private final PlantationItem combination;
-
-        /**
-         * Default constructor.
-         *
-         * @param position    the position.
-         * @param combination the item combination data.
-         */
-        private PlantationSoilPosition(final BlockPos position, final PlantationItem combination)
+        @Override
+        public void deserialize(final @NotNull FriendlyByteBuf buf)
         {
-            this.position = position;
-            this.combination = combination;
+            super.deserialize(buf);
+            maxConcurrentPlants = buf.readInt();
         }
 
-        /**
-         * Get the position.
-         *
-         * @return the position.
```

Comment:
reorg of methods makes this unreadable

Reply:
This specific class is actually not even reorganized, there's just *that* many changes. Almost nothing of the original class survived.